session/payment/factory/payments.go

```diff
@@ -51,21 +57,30 @@ func noopPaymentIssuerFactory(initialState promise.State,
 }
 
 func paymentIssuerFactory(signerFactory identity.SignerFactory) func(
-	initialState promise.State,
+	initialState promise.PaymentInfo,
+	paymentDefinition dto.PaymentPerTime,
 	messageChan chan balance.Message,
 	dialog communication.Dialog,
 	consumer, provider identity.Identity) (connection.PaymentIssuer, error) {
 	return func(
-		initialState promise.State,
+		initialState promise.PaymentInfo,
+		paymentDefinition dto.PaymentPerTime,
 		messageChan chan balance.Message,
 		dialog communication.Dialog,
 		consumer, provider identity.Identity) (connection.PaymentIssuer, error) {
 
 		bl := balance.NewListener(messageChan)
 		ps := promise.NewSender(dialog)
 		issuer := issuers.NewLocalIssuer(signerFactory(consumer))
-		tracker := promise.NewConsumerTracker(initialState, consumer, provider, issuer)
-		payments := payment.NewSessionPayments(messageChan, ps, tracker)
+		tracker := promise.NewConsumerTracker(promise.State{
```

Comment:
Looks like too much glue logic in factory, such fat factories demands to be tested.
Or can we move them out somehow?

Reply:
I've added a test for the only piece of logic we had there.